pkgs/development/compilers/elm/default.nix

```diff
@@ -45,7 +45,7 @@ let
       let hlib = haskell.lib;
           elmRelease = import ./packages/release.nix { inherit (self) callPackage; };
           elmPkgs' = elmRelease.packages;
-          elmPkgs = elmPkgs' // {
+          elmPkgs = lib.mapAttrs (name: value: hlib.disableSharedExecutables value) (elmPkgs' // {
```

Comment:
Is this correct? I'm not a user of the elm packages, but this line looks wrong to me: it only disable shared executables for "outside" users of the elm packages, but if any package in the elmPkgs set depends on another one, it will get the shared executable version (b.c. the fixpoint set isn't changed. the modification is applied after the fixpoint).

For example, if we had this situtation:

```
elmPkgs = lib.mapAttrs lib.mapAttrs (name: value: hlib.disableSharedExecutables value) ({
  elm-exe-a = ... derivation depending on self.elm-exe-b
  elm-exe-b = ... some executable ...
})
```

then `elm-exe-a` will get the shared version of `elm-exe-b`, but `elmPkgs.elm-exe-b` will refer to the non-shared version. Maybe it does not matter in this case though because no elm package depends on executables from another one? Or is it actually meant to be only for the "outside"? (why?)

Reply:
You're right, need to properly override the scope. Couldn't find an easy way to do that without explicitly enumerating all elm packages.

Reply:
Fixed, this will do a few more rebuilds but it's a proper override.